### ISSUE TITLE
Discussion: just a thought

### ISSUE THREAD

**Issue by auahub** [Microsoft/xaml-standard#222]
I noticed that most contributors are suggesting a new controls or features, I think the scope of this project is to agree on which controls to standardize from different platforms not to suggest a new feature, like for example we should discuss that we take stackpanel from UWP or stacklayout from xamarin. am I right here or wrong?

Thank you for contributing to XAML Standard! We are at the **_very early stages_** of a long term collaboration towards a unified XAML vocabulary that all XAML developers can comfortably rely on to target all your desired end points.

## All submissions
- [ ] Have you read the [principles](https://github.com/Microsoft/xaml-standard/blob/staging/docs/reviewboard.md#principles)?
- [ ] Have you checked that your post fits the [nature of proposals](https://github.com/Microsoft/xaml-standard/blob/staging/docs/proposalsfaq.md#nature-of-a-proposal)?
- [ ] Have you checked to ensure that there aren't other open [issues](https://github.com/Microsoft/xaml-standard/issues) for the same topic?
- [ ] Have you followed the [contributor code of conduct](https://dotnetfoundation.org/code-of-conduct)?

## If you are posting an API for XAML standard consideration:
- Add a brief overview of what you are proposing and why it belongs in XAML Standard (in keeping with the principles of the effort).
- Provide a clear usecase/scenario that describes how this API benefits in sharing cross-platform UI code, preferably a scenario from your experience or application that clearly illustrates why a control/API would benefit from being standardized.
- Add a brief example in XAML syntax for your proposal.
- Provide links to issues/discussions where this proposal has been talked about as well as any other pointer to where this originated or got discussed that could describe the motivation.

For discussions that are not concrete proposals, please prefix title with "Discussion" so we can categorize accordingly.

**Comment by insinfo** [comment]
Yes I agree that the initial thinking is an alignment between the UWP, WPF and Xamarin platforms, plus the standard can not stop here, the XAML standard has to continuously evolve, this means that the standard will add or set new features over time , this includes specification of new controls and standardized APIs that will have to be implemented by the XAML standard supporters.

I believe that the XAML Standard has to keep advancing

**Comment by touseefbsb** [comment]
XAML standard should be like .NET STANDARD where eventually xaml and c sharp can take over all possible platforms, and we can write one app with c sharp and xaml which might run on android, ios, uwp, linux, mac os and web as well. xaml standard and .net standard should become the new standard way of building software.

**Comment by insinfo** [comment]
@touseefbsb I completely agree and subscribe below, this would be great if we could write truly multiplatform native applications with XAML and C #

**Comment by insinfo** [comment]
And for this to be possible it is necessary to assemble the pieces of the great puzzle that is here in github, the ecosystem XAML and C#

https://github.com/AvaloniaUI/Avalonia
https://github.com/yuvaltz/Granular
https://github.com/sunnycase/MPF
https://github.com/lrz/mono-wasm
https://github.com/SteveSanderson/Blazor
http://www.mono-project.com/news/2017/08/09/hello-webassembly/

**Comment by touseefbsb** [comment]
@insinfo yes all these pieces can be assembled but the project of this level should be directly handled by Microsoft itself and backed up by the community of c# and xaml developers to make it a success. Majority of c#/xaml developers dont really like to code in JS and the infinite number of frameworks out there. Alternatively we should make .net and xaml standard so strong and performant that we can create every typeof software and websites with just these 2 languages without needing so many frameworks.

**Comment by insinfo** [comment]
@touseefbsb I fully agree, especially with this part

> "create every typeof software and websites with just these 2 languages without needing so many frameworks."

**Comment by auahub** [comment]
I also agree that the standard cannot stop here, but I think everything should be done step by step 1st standardization then further development.

while going through the discussion I have got another idea I will open a new discussion issue, maybe  microsoft engineers will be convinced by it.

________________________________

**Comment by touseefbsb** [comment]
yeah convincing microsoft regarding all these ideas is most important thing, bcz this level of technology has to be directly engineered by microsoft itself

**Comment by touseefbsb** [comment]
should we open a user voice as well regarding this? @auahub  , @insinfo

**Comment by Mike-EEE** [comment]
FWIW @touseefbsb you can add your votes and feedback to this idea here:
https://visualstudio.uservoice.com/forums/121579-visual-studio-ide/suggestions/10027638-create-a-ubiquitous-net-client-application-develo

Most popular idea on Visual Studio's UserVoice ATM, passing 8k votes as of last week.

**Comment by touseefbsb** [comment]
@Mike-EEE  Thanks a lot. Thatis the perfect uservoice for this requirement it has too many votes, I hope microsoft acts quickly on this and make this a reality ASAP.

**Comment by auahub** [comment]
@Mike-EEE thats a great suggestion
but the problem as I think it is too complicated for microsoft to start with and I think my idea in issue #228 may be simpler to start with, then try to reach the the ultimate goal mentioned in the user voice.

**Comment by insinfo** [comment]
@Mike-EEE I voted for your UserVoice

**Comment by Mike-EEE** [comment]
Thanks all.  We're actually a lot closer to this reality than you might think.  Once Mono officially adds support for WebAssembly (and they are [on their way](https://github.com/mono/mono/pull/5924)), we'll be able to reach the Big Four: Droid, iOS, Windows, and Web.  The roads will be paved and it will be a matter of putting the cars on them to drive.  As such, when discussing models (cars) here, don't forget about [Noesis](http://www.noesisengine.com/) which is my top vehicle of choice ATM. 😎

**Comment by weitzhandler** [comment]
This repo is sleeping.
Looks like MS pays to it very little attention.
Last commit was November '17.

IMHO, MS has the power to change the multi-platform development face of the entire world if they provide a complete C#+XAML solution for all platforms (UWP+Web), but then MS also has to provide easy tooling to make development time less painful, and that includes:
- Look like Xamarin isn't planning to implement the very essential controls it lacks, such as `ItemsControl`, horizontal `ListView`,  `CheckBox`, `RadioButton` etc., and that's because their agenda says only controls native to all platforms are implemented (how stupid).
- Easy tooling for XAML-based clients and server side interaction (such as [Breeze#](https://github.com/Breeze/breeze.sharp) or [TrackableEntities](https://github.com/TrackableEntities/trackable-entities), but officially maintained.
- Client side entity generators
- XAML designers or at least previewers for ALL platforms (Xamarin.UWP <s>and XAML web</s> included).
- Easier ways to deploy to all platforms in one click ([App Center](https://www.visualstudio.com/app-center/) does a decent job).

**Comment by touseefbsb** [comment]
@weitzhandler  the features you just described, please make a proper uservoice for that and post the link here so that we all can contribute and vote for that. we have to work hard in order for this to become a reality and I agree with all ur points that we need those crucial controls. c# and xaml really need to target all big 4.

**Comment by insinfo** [comment]
@weitzhandler Undoubtedly, we need better, more competitive, smarter, and more complete development tools, integrating artificial intelligence and node-based visual programming with UI and UML designer integration, so we can build the next generation of intelligent software that interprets the voice in natural conversation, touch, vision among other means of human machine interaction.

**Comment by weitzhandler** [comment]
@insinfo Sky is the limit, but before we try reaching the sky, I'm merely requesting the basic functionality we were used to in WinForms and Silverlight+RIA services to be back, which is included in the aspects I pointed out [before](https://github.com/Microsoft/xaml-standard/issues/222#issuecomment-357454799).

**Comment by weitzhandler** [comment]
@touseefbsb 
I don't really believe in UserVoice. The reason is that in UserVoice, only the first suggestions get upvoted and interesting, while all the new ones never get anyone's attention and will remain abandoned and unread forever. UserVoice is really an overrated service.
But anyway, [here](https://xamarin.uservoice.com/forums/144858-xamarin-platform-suggestions/suggestions/32944858-implement-elementary-controls-in-xamarin-forms) you go!

@insinfo Sky is the limit, but before we try reaching the sky, I'm merely requesting the basic functionality we were used to in WinForms and Silverlight+RIA services to be back, which is included in the aspects I pointed out [before](https://github.com/Microsoft/xaml-standard/issues/222#issuecomment-357454799).

**Comment by touseefbsb** [comment]
@weitzhandler  I knw it is overrated but we dont have any other choice :P

**Comment by touseefbsb** [comment]
hhaha I wish I could :P @weitzhandler
